src/io/output_data.cc

```diff
@@ -93,6 +94,57 @@ void OutputData<Value>::print_all(ostream &out_stream)
 }
 
 
+/// Prints the whole data vector into stream.
+template <class Value>
+void OutputData<Value>::print_binary_all(std::ostream &append_stream, std::ostream &out_stream)
+{
+    print_binary_inner(append_stream, out_stream);
+}
+
+
+template<class Value>
+template<class Q>
+void OutputData<Value>::print_binary_inner(ostream &append_stream, ostream &out_stream,
+		typename std::enable_if<std::is_floating_point<typename Q::element_type>::value >::type* )
+{
+	// variables storing range of data array
```

Comment:
Rather compute range of OutputData in separate method, keep the print_all methods simple, doing only ou